File: Flip Bits - GFG/flip-bits.py
#User function Template for python3

class Solution:
    def maxOnes(self, arr, n):
        
        # Your code goes here
        max_ending_here = 0
        max_so_far = 0
        ones_count = 0
    
        for i in range(n):
            max_ending_here += 1 if arr[i] == 0 else -1
            max_so_far = max(max_so_far, max_ending_here)
            ones_count += arr[i]
            max_ending_here = max(0, max_ending_here)
        
        if max_so_far == 0 and ones_count == 0:
            return n - 1
        
        return max_so_far + ones_count
                    
                    
       


#{ 
 # Driver Code Starts
#Initial Template for Python 3

def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        ob=Solution()
        print(ob.maxOnes(a, n))

        T -= 1


if __name__ == "__main__":
    main()



# } Driver Code Ends

File: Flip Bits - GFG/flip-bits.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
    public:
    int maxOnes(int a[], int n)
    {
        // Your code goes here
        int co0 =0, co1= 0;
        int count1 =0;
        int fc0 = 0, fc1 = 0;
        for(int i=0;i<n;i++) {
            if(a[i] == 1) 
            count1++;
        }
        int maxv = 0;
        for(int i=0;i<n;i++) {
            if(a[i] == 0) {
                co0++;
            } else {
                co1++;
            }
            if(co0 <= co1) {
                co0 = 0;
                co1 = 0;
            }
            //cout<<co0<<" "<<co1<<"\n";
            maxv = max(maxv, co0 - co1);
            if(maxv == co0 -co1) {
                fc0 = co0;
                fc1 = co1;
            }
        }
        return count1 + (fc0 - fc1);
        // for each subarray co0 - co1 is added to array 
        /// each subarray should start and end with 0
    }
};


//{ Driver Code Starts.
int main()
{
    int t; cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        int a[n+5];
        for(int i=0;i<n;i++)
            cin>>a[i];
        Solution ob;
        cout<< ob.maxOnes(a, n) <<endl;
    }
    return 0;
}

// } Driver Code Ends